src/container-manager/index.ts

```diff
@@ -1,41 +1,141 @@
+        this.rskNodeEnvironment = await this.startDockerComposeModule(path.resolve(__dirname, 'docker-files/rsk-node'));
+        await waitForHealthCheck({
+            command: 'healthCheck',
+            expectedResult: 'NODE ONLINE',
+            container: this.rskNodeEnvironment.getContainer('rsk-node')
+        });
+        console.debug('Started rsk node successfully');
+        return Promise.resolve(this.rskNodeEnvironment);
     }
 
-    public async startupLuminoNode(nodeConfig: SetupNode): Promise<LuminoNode> {
-        console.log('Setup Lumino', nodeConfig);
-        //TODO: pending add dockerfile to testcontainers
+    public async startExplorer(): Promise<StartedDockerComposeEnvironment> {
+        console.debug('Starting lumino explorer...');
+        this.explorerEnvironment = await this.startDockerComposeModule(path.resolve(__dirname, 'docker-files/lumino-explorer'));
+        await waitForHealthCheck({
+            command: 'healthCheck',
+            expectedResult: 'NODE ONLINE',
+            container: this.explorerEnvironment.getContainer('lumino-explorer')
+        });
+        console.debug('Started lumino explorer successfully');
+        return Promise.resolve(this.explorerEnvironment);
+    }
+
+    public async startNotifier(instanceNumber: number): Promise<StartedGenericContainer> {
         throw new Error("NOT_IMPLEMENTED");
     }
 
+    private async startDockerComposeModule(buildContext: string): Promise<StartedDockerComposeEnvironment> {
+        return await new DockerComposeEnvironment(
+            buildContext,
+            `docker-compose.yml`
+        ).up();
+    }
+
+    public getContainer(
+        containerName: 'rsk-node' | 'lumino-explorer' | 'mongo-db' | 'lumino-node',
+        luminoNodeName?: string,
+        scaleNumber?: number): StartedGenericContainer | StartedTestContainer | undefined {
+        switch (containerName) {
+            case 'rsk-node':
+                return this.rskNodeEnvironment?.getContainer(containerName);
+            case 'lumino-explorer':
+            case 'mongo-db':
+                return this.explorerEnvironment?.getContainer(containerName);
+            case "lumino-node":
+                return this.luminoNodes.find(node => node.name === luminoNodeName)?.container;
+        }
+    }
+
+    public async startupLuminoNode(nodeConfig: SetupNode): Promise<LuminoNode> {
+
+        console.debug('Creating lumino node', nodeConfig);
+
+        let luminoNodeContainer: GenericContainer = new GenericContainer('lumino-testing-lumino-node-image:latest');
+
+        luminoNodeContainer = luminoNodeContainer.withNetworkMode(DOCKER_NETWORK_NAME);
+
+        if (nodeConfig.enableHub) {
+            console.debug(`Setup Hub Mode to lumino node ${nodeConfig.name}`);
+            luminoNodeContainer = luminoNodeContainer.withEnv('HUB_MODE', 'enabled');
+        }
+
+        if (nodeConfig.tokens) {
+            for (const tokenSetup of nodeConfig.tokens) {
```

Comment:
Maybe
```suggestion
        nodeConfig.tokens?.forEach(tokenSetup => {
```
Could help to reduce the identation levels.

Reply:
Can't do that since .forEach requires final constants instead of variables and luminoNodeContainer it's a variable that changes inside the loop.